api/controllers/StatsController.js

```diff
@@ -6,6 +6,40 @@ dayjs.extend(isBetween);
 /////////////
 // Helpers //
 /////////////
+/**
+ * Find the season in which a given timestamp occurred using binary search
+ * @param {*} sortedSeasons - Seasons sorted (OLDEST FIRST)
+ * @param {*} timeStamp - Desired time in millis since epoch
+ * @returns the season | null if none is found
+ */
+function getSeasonFromTimeStamp(sortedSeasons, timeStamp) {
+  let currentIndex;
+  let minIndex = 0;
+  let maxIndex = sortedSeasons.length - 1;
+  // Sanity check in case list is unsorted or start & endTimes are out of whack
```

Comment:
I need a little more context regarding this bit of logic, when and how do they get out of whack?

Reply:
So realistically it shouldn't. So first the purpose of this function is to identify in which season a given timestamp occurred ie which season was this game or season in? 

The function should return the relevant season if there is one and `null` otherwise. To do that, it makes several assumptions, namely that every season's time `startTime` is before its `endTime` and that the `sortedSeasons` is sorted in descending chronological order.

That should always be the case unless one or more of the seasons get defined incorrectly in the database or if this function gets called in an additional place with incorrectly sorted args. If either of those weird situations happen, we want it to just return `null` to ensure we never infinitely loop. That said, my "sanity check" here doesn't look like it would ever fire because the conditional 

```
if (minIndex === maxImdex) {
  return null;
}
```
should always happen first (since it takes log_2(sortedSeasons.length) loops for min and max index to meet, but hitting this `numTries` limit requires as many loops as the length of the list, which we can't hit before the other check. Which means the while loop could just be `while (true)`, but even though I can't imagine a way for that to infinite loop, that still makes me nervous